### ISSUE TITLE
What does `--tccdb` do exactly and how to use it?

### ISSUE THREAD

**Issue by janpio** [apache/cordova-paramedic#71]
This is about https://github.com/apache/cordova-paramedic#--tccdb-optional

> **--tccDb (optional)**
> iOS only parameter. The path to the sample TCC DB file, with permissions, to be copied to the simulator.
> `cordova-paramedic --platform ios --plugin cordova-plugin-contacts --tccDbPath tcc.db`

Was added to the codebase with this commit: https://github.com/apache/cordova-paramedic/commit/739c4326ba8c5014d4776e53b28e6da817c85cfb

But doesn't seem to be used anywhere in our repositories:
https://github.com/search?q=org%3Aapache+tccDb&type=Code

---

sujan [5:15 PM]
What is a "TCC DB file"?
it has to do with iOS, permissions, maybe contacts
Was added to paramedic here: https://github.com/apache/cordova-paramedic/commit/739c4326ba8c5014d4776e53b28e6da817c85cfb
"Apple’s TCC privacy database"

dpogue [5:16 PM]
Seems to be a database of apps and their permissions

sujan [5:17 PM]
> The path to the sample TCC DB file, with permissions, to be copied to the simulator.

**Comment by janpio** [comment]
Cordova Medic (archived) seems to have used this file as well for something:
https://github.com/apache/cordova-medic/blob/d3efbf1b888b49d43a1d99b5b646409e87f963d3/JENKINS.md (at the end of the document)
https://github.com/apache/cordova-medic/tree/d3efbf1b888b49d43a1d99b5b646409e87f963d3/lib/tcc

**Comment by oliversalzburg** [comment]
From reading [accessing TCC.db without privileges](https://applehelpwriter.com/2018/07/06/accessing-tcc-db-without-privileges/), [About System Integrity Protection on your Mac](https://support.apple.com/en-us/HT204899) and [Editing the TCC.db with sqlite3](https://www.jamf.com/jamf-nation/discussions/23921/editing-the-tcc-db-with-sqlite3), I would assume this feature was intended to pre-set certain Accessibility features for the app under test in the simulator.

**Comment by janpio** [comment]
What would be the practical use of this in the context of paramedic, a tool to build an app an run tests included in it?

**Comment by oliversalzburg** [comment]
I guess if you deploy the camera plugin (for example), the app will need camera privileges, which would have to be manually confirmed. By injecting this file, the prompt can be skipped. I'm still just assuming :)

**Comment by janpio** [comment]
Yeah, something like that - but I couldn't find the expected file in any of our repositories.

**Comment by oliversalzburg** [comment]
As I understand the source, the file is just copied into the simulator folder on disk (maybe I didn't read that right). So I assumed you would run it manually once and copy the file out of the folder for later re-use.

**Comment by janpio** [comment]
nah, setPermissions is in the normal code execution of paramedic:
https://github.com/apache/cordova-paramedic/blob/dfed2c716bfaf1c9bfde2fba5c073d8e0685d7af/lib/paramedic.js#L240-L245

Then it either copies the file if it doesn't exist yet, or update the entries:
https://github.com/apache/cordova-paramedic/blob/dfed2c716bfaf1c9bfde2fba5c073d8e0685d7af/lib/ParamediciOSPermissions.js
